src/routes/blogs.rs

```diff
@@ -100,25 +101,29 @@ pub fn new(rockets: PlumeRocket, _user: User) -> Ructe {
 #[get("/validate/<validation_id>")]
 pub fn domain_validation(
     validation_id: String,
-    valid_domains: State<HashMap<&str, Instant>>,
-) -> Result<String, String> {
-    let value = valid_domains.inner().get(validation_id.as_str());
+    valid_domains: State<Mutex<HashMap<&str, Instant>>>,
+) -> Status {
+    let mutex = valid_domains.inner().lock();
+    let mut validation_map = mutex.unwrap();
+    let validation_getter = validation_map.clone();
+
+    let value = validation_getter.get(validation_id.as_str());
     if value.is_none() {
-        return Err(String::from("404"));
+        return Status::new(404, "validation id not found");
     }
 
     // we have valid id, now check the time
     let valid_until = value.unwrap();
     let now = Instant::now();
 
     // nope, expired (410: gone)
-    if now.checked_duration_since(*valid_until).unwrap().as_secs() > 0 {
-        valid_domains.inner().remove(validation_id.as_str());
-        return Err(String::from("410"));
+    if now.duration_since(*valid_until).as_secs() > 0 {
+        validation_map.remove(validation_id.as_str());
+        return Status::new(410, "validation expired");
```

Comment:
Same as above

Reply:
```
error[E0308]: mismatched types
   --> src/routes/blogs.rs:113:28
    |
113 |         return Status::new(Status::NotFound, "validation id not found");
    |                            ^^^^^^^^^^^^^^^^ expected u16, found struct `rocket::http::Status`
    |
    = note: expected type `u16`
               found type `rocket::http::Status`

error[E0308]: mismatched types
   --> src/routes/blogs.rs:123:28
    |
123 |         return Status::new(Status::Gone, "validation expired");
    |                            ^^^^^^^^^^^^ expected u16, found struct `rocket::http::Status`
    |
    = note: expected type `u16`
               found type `rocket::http::Status`
```

Reply:
Have you tried simply
```suggestion
        return Status::Gone;
```
?

Reply:
No, cuz I wanted to give a specific response.

Reply:
```suggestion
        return Custom(Status::Gone, "validation expired");
```
imported from rocket::response::status::Custom
Your version is sending a standard error code with a non standard text, some implementations don't handle this well. This return the corresponding "error" text, and set the body of the result to the text you provide